### ISSUE TITLE
Todos & Question

### ISSUE THREAD

**Issue by aight8** [cristiandley/react-gl-maps#6]
Hey I want to contribute to this library to make a stable library.

I have some points as notes:
- [x] Remove depricated methods and warnings -> major version release https://github.com/cristiandley/react-gl-maps/pull/12/commits/cd2038ddd5889dcbf6534420fe2a73aa550fe74c
- [x] Update dependencies https://github.com/cristiandley/react-gl-maps/pull/12/commits/78e2f3c5c6c3e4fe9ddd8f3189430f85d556e5a3
- [ ] Fix babel errors.
- [ ] Fix eslint errors.
- [ ] Ignore `/lib`. https://github.com/cristiandley/react-gl-maps/pull/12/commits/eea88e831c75f28707c1fb5ac1077c8884b784ec
- [ ] Fix package build.
- [ ] Proper Errors.
- [ ] Establish a package loading architecture.
- [ ] Fix package loading architecture. https://github.com/cristiandley/react-gl-maps/issues/10
- [ ] Update modules.
- [ ] Improve documentation.
- [ ] Implement valid testing.
- [ ] Remove obsolete code.
- [ ] Fix examples.

And I have a question related to the contribution, especially babel. I see often that in projects there is no .babelrc or a babel entry in package.json, but I don't understand why. Babel need presets and plugins to compile a specific project correctly - furthermore there are more dependencies which are not in the dev-dependency list (transform-decorators-legacy etc.). So I figured out how babelrc should look like:

``` JSON
{
  "plugins": [
    "transform-decorators-legacy",
    "transform-class-properties"
  ],
  "presets": [
    "react",
    "es2015",
    "stage-0"
  ]
}
```

**Comment by rewop** [comment]
@aight8 great analysis. 

About this issue I agree with your points. Only one comment, why do you want to change scriptjs with loadjs? I would like to understand the value it would add.

`package.json` should be definitely updated. 
An additional point to your list is the `lib` folder as discussed in #5. It shouldn't be committed in the repository.

**A side note**
I think the development, and versioning of this repo, as well as the list of main contributors must be discussed properly. 

This repo was born to solve the problem that the pr https://github.com/tomchentw/react-google-maps/pull/224, as well as others, would never be merged. so @cristiandley decided to start this new repo and publish it on npm, to be able to make use of the performances issue. It would be great to know also what @tomchentw thinks about it.

Maybe we should open an issue to discuss this. @cristiandley

**Comment by cristiandley** [comment]
@aight8 good points, i love watching people following the fork!

_A bit of history:_
@rewop I tried to contact @tomchentw and waited two months before becoming crazy about the fact that the repo was dead, but there is no life signals from him. I want this package to be a great one ... well... already is but thats is a good and a bad thing, a lot of people having great ideas and non of them being merge. So i was in the corner of having to implement experimental things (that i still testing, like performance over real time markers) but not having any officiall support or way to contact the author.

@aight8  @rewop  @jwdotjs @vinceprofeta  There is a lot of work to do, lets open an issue and promote a road map. Wouldn't be cool ?

I would like also to create a group and give them administration permissions, so im not the only one being able to merge working ideas or managing the path of this repo (we all have a life and activities and i think this repo deserves a dev group).

Basically i want a team... that way we can promote the future of this repo.

i wrote too much ? im sorry :)

Thanks guys for being awesome!

**Comment by aight8** [comment]
Ah sorry guys I don't see babelrc, it's there but it's the old one for babel 5. I already made an upgrade to babel 6 and will PR it with other changes.

Thank you for the explanation, I think we can make an awesome library since google maps is really something users use - and yes in my startup too and I'm totally need a flexible map solution there - and of course with react :)
The idea with the permission is good.

A roadmap would be very good. We can discuss about everything than we can make a roadmap with tasks etc., alright? Until the repository is not to popular we have time to make a stable fundament.

What do you guys think, where should we start and what's your experience with this library?

**Comment by rewop** [comment]
@cristiandley I would love to help to maintain the module. At the moment though I am very busy at work. Hopefully soon I will be more available. But definitely consider me in.

Also I suggest to come out with a proper roadmap, starting from updating the modules, fixing the build of the module, and simplifying the code (in my opinion too much not needed abstraction). We could keep discussing it here.

**Comment by aight8** [comment]
@rewop about loadjs: Only because of the failure handling.

```
LoadJS is based on the excellent $script library by Dustin Diaz.
We kept the behavior of the library the same but we re-wrote the code from scratch to add support for success/failure callbacks and to optimize the library for modern browsers.
LoadJS is 537 bytes (minified + gzipped).
```

**Comment by cristiandley** [comment]
In my opinion simple is better...  i want clear code. Nothing fancy... after having a solid base we could grow.

First we should fix major bugs to have a clear view of what we have....

All the PR's from now should at least leave an example or a brief comment of what they improve. In @tomchentw there is a lot of PR's without a clear explanation like "Update Drawer" and they commited 10 files changes.

I edited your comment @aight8

**Comment by tomchentw** [comment]
Hi guys,

First, I want to apologize for not being active on maintaining the module, `react-google-maps`. The main reason is that I don't have any production projects using it for a year and more. Plus, the overhead of the issues and lack of documentations making it even harder to keep issues clean.

I should have asked the great guys like @cristiandley and others for help in order to keep the community healthy. So, here's what I'm gonna to do,
1. Add "Call for maintainers" section in `README.md`.
2. Review current PRs ASAP, merge it if possible and release minor/patch version
3. Create guidelines for maintaining the repo/commit messages/release process……etc
4. Add contributors to the repository. @cristiandley and anyone else?
5. Review on the fork (react-gl-maps) to see if we can merge them together.

Please let me know what you think guys. Also, feel free to reach me out via my email (you can see it on my GitHub profile) if I'm not responding. Sometimes email of issues get lost in my inbox.

Thank you guys for all the hard work!

**Comment by aray12** [comment]
Hey so I have read through some of the comments here and on react-google-maps, but just want to confirm. Is this dead and the use of react-google-maps encouraged?
